### PR TITLE
feat(web-iconics) : implement button component 

### PR DESCRIPTION
**Pull request by zerihunMoges**
create a button component

### CODE REVIEW

**Comment by YiseBoge** [comment]
please rename the PR name to follow conventional commits (because that is what will be used when using squash and merge by default)

**Comment by zerihunMoges** [comment]
> please rename the PR name to follow conventional commits (because that is what will be used when using squash and merge by default)
> 
> And please put in screenshots and description of what you have implemented in the description

button

I created a generic button component that can be modified and also accepts optional icon props if needed

**Review by Adona12 — CHANGES_REQUESTED** [review]
The branch is out of date so make sure to rebase

**Comment by zerihunMoges** [comment]
> The branch is out of date so make sure to rebase

done

**Comment by zerihunMoges** [comment]
> The branch is out of date so make sure to rebase

Screenshot 2023-04-26 041350

File: Web/starter_project_iconics/components/common/Button.tsx [hunk 1]
```diff
@@ -15,14 +15,23 @@ export interface ButtonProps extends React.ComponentProps<'button'> {
...
-function Button({label,
-className,
-...props}: ButtonProps) {
+function Button({ label,
+  className, startIcon, endIcon, outline,
+  ...props }: ButtonProps){
   return (
-    <button {...props} className={clsx( 'px-4 py-2 m-2 text-white bg-[#264FAD] rounded-md cursor-pointer', className)}>{label}  </button>
+
+    <button {...props} className={clsx( outline? 'outline outline-2 outline-primary text-primary bg-white hover:bg-primary hover:text-white' : 'text-white bg-primary hover:bg-blue-600', 'flex justify-center items-center gap-x-3 px-8 py-3 m-2 text-md  rounded-md cursor-pointer', className)}>{startIcon && startIcon} {label} {endIcon && endIcon} </button>
```
Comment (Adona12) [inline Web/starter_project_iconics/components/common/Button.tsx:33]: We don't need to have the hover propery

**Review by Adona12 — APPROVED** [review]
LGTM

**Pull request merged**